### PR TITLE
Update miscellaneous libraries

### PR DESCRIPTION
**Pull request by TacoTheDank**
#### What is it?
- [ ] Bugfix (user facing)
- [ ] Feature (user facing)
- [x] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
- Update Jsoup 1.16.1 -> 1.16.2 ([changelog](https://jsoup.org/news/release-1.16.2))
- Update OkHttp 4.11.0 -> 4.12.0 ([changelog](https://square.github.io/okhttp/changelogs/changelog_4x/#version-4120))
- Update ACRA 5.10.1 -> 5.11.3 ([changelog](https://github.com/ACRA/acra/releases))
- Update RxJava 3.1.6 -> 3.1.8 ([changelog](https://github.com/ReactiveX/RxJava/releases))
- Update Prettytime 5.0.6.Final -> 5.0.7.Final ([changelog](https://github.com/ocpsoft/prettytime/compare/5.0.6.Final...5.0.7.Final))
- Update Mockito 4.0.0 -> 5.6.0 ([changelog](https://github.com/mockito/mockito/releases))
- Update AssertJ 3.23.1 -> 3.24.2 ([changelog](https://github.com/assertj/assertj/releases))
- Update Kotlin 1.8.22 -> 1.9.10 ([changelog](https://kotlinlang.org/docs/whatsnew19.html))
- Update AGP 8.1.1 -> 8.1.2 ([changelog](https://developer.android.com/studio/releases/fixed-bugs/studio/2022.3.1#android-studio-giraffe-|-2022.3.1-patch-2))

#### APK testing
The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR. You can find more info and a video demonstration [on this wiki page](https://github.com/TeamNewPipe/NewPipe/wiki/Download-APK-for-PR).

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by AudricV — COMMENTED** [review]
Thanks!

In a similar way to jsoup with its `README.md` and `CHANGES` files, RxJava is now adding `LICENSE` and `COPYRIGHT` files into its JARs: see https://github.com/ReactiveX/RxJava/blob/e6f152df7e2f2061435ee8ac3f6ed82cf34e3c58/build.gradle#L90-L94. Could you remove them?

**Comment by TacoTheDank** [comment]
> In a similar way to jsoup with its `README.md` and `CHANGES` files, RxJava is now adding `LICENSE` and `COPYRIGHT` files into its JARs: see https://github.com/ReactiveX/RxJava/blob/e6f152df7e2f2061435ee8ac3f6ed82cf34e3c58/build.gradle#L90-L94. Could you remove them?

I checked the built artifact, and it seems that only the COPYRIGHT file was there. LICENSE wasn't in the META-INF.

File: app/build.gradle [hunk 1]
```diff
@@ -98,7 +98,8 @@ android {
         resources {
             // remove two files which belong to jsoup
             // no idea how they ended up in the META-INF dir...
```
Comment (AudricV) [inline app/build.gradle:100]: Please also add a comment adding that you are removing the COPYRIGHT file of RxJava here.

**Review by AudricV — CHANGES_REQUESTED** [review]
Almost good, thank you!

Have any dependency updates been released in the meantime? If so, could you apply them?

File: app/build.gradle (hunk 1 above)
Comment (TacoTheDank) [reply app/build.gradle:100]: Gotcha, I thought I did but I guess I forgot to when committing.

**Comment by TacoTheDank** [comment]
> Have any dependency updates been released in the meantime? If so, could you apply them?

Done. I didn't do Kotlin though, I want to update Ktlint first in another PR.

**Comment by Stypox** [comment]
I pushed more updates:
- `androidx.fragment:fragment-ktx` 1.6.1 -> 1.6.2 ([changelog](https://developer.android.com/jetpack/androidx/releases/fragment))
- `androidx.media:media` 1.6.0 -> 1.7.0 ([changelog](https://developer.android.com/jetpack/androidx/releases/media))
- `com.google.android.material:material` 1.9.0 -> 1.11.0 ([changelog](https://github.com/material-components/material-components-android/releases))
- Android Gradle Plugin 8.1.4 -> 8.2.0 ([changelog](https://developer.android.com/build/releases/gradle-plugin))
- Gradle 8.1.1 -> 8.2.0

**Pull request merged**